src/batch.jl

```diff
@@ -0,0 +1,210 @@
+struct BatchIntegrand{F,Y,X}
```

Comment:
I think it might be more conventional to have `X,Y` rather than `Y,X`, and similarly change the argument order to `x,y` below?

Reply:
Makes sense.